CyberSource/api/batch_upload_with_mtls_api.py

```diff
@@ -0,0 +1,212 @@
+"""
+Batch Upload with mTLS API for CyberSource.
+
+This module provides functionality for batch uploads to CyberSource using mTLS authentication,
+supporting both file-based and in-memory operations with various configuration options.
+"""
+
+from pathlib import Path
+from typing import Optional
+
+import CyberSource.logging.log_factory as LogFactory
+from CyberSource.utilities.pgpBatchUpload.mutual_auth_upload import MutualAuthUpload
+from CyberSource.utilities.pgpBatchUpload.pgp_encryption import PgpEncryption
+
+
+def _validate_paths(paths: list) -> None:
+    """
+    Validate that all specified paths exist.
+
+    Args:
+        paths: List of tuples containing (path, description)
+
+    Raises:
+        FileNotFoundError: If any of the paths don't exist
+    """
+    for path, description in paths:
+        if path is None:
+            continue  # Skip validation for None paths
+        if not path:
+            raise ValueError(f"{description} path is required")
+        if not Path(path).exists():
+            raise FileNotFoundError(f"{description} not found: {path}")
+
+
+class BatchUploadWithMTLSApi:
+    """
+    A class for handling batch uploads to CyberSource using mTLS authentication.
+
+    This class provides methods for encrypting and uploading batch files to CyberSource
+    using PGP encryption and mutual TLS authentication. It supports various authentication
+    methods including separate key/cert files.
+
+    Attributes:
+        logger: Logger instance for logging operations and errors
+        pgp_encryption: PgpEncryption instance for handling PGP encryption
+        mutual_auth_upload: MutualAuthUpload instance for handling mTLS uploads
+    """
+
+    _end_point = "/pts/v1/transaction-batch-upload"
+
+    def __init__(self, log_config=None):
+        """
+        Initialize the BatchUploadWithMTLSApi.
+
+        Args:
+            log_config: Optional configuration for the logger
+        """
+        self.logger = LogFactory.setup_logger(self.__class__.__name__, log_config)
+        self.pgp_encryption = PgpEncryption(log_config)
+        self.mutual_auth_upload = MutualAuthUpload(log_config)
+
+    def upload_batch_api_with_separate_key_and_cert_file(
```

Comment:
upload_batch_api_with_separate_key_and_cert_file to upload_batch_api_with_key_and_cert_file